modules/build/src/main/scala/scala/build/internal/NativeBuilderHelper.scala

```diff
@@ -0,0 +1,65 @@
+package scala.build.internal
+
+import java.math.BigInteger
+import java.security.MessageDigest
+
+import scala.build.Build
+import scala.scalanative.{build => sn}
+
+case object NativeBuilderHelper {
+
+  private def resolveProjectShaPath(nativeWorkDir: os.Path) = nativeWorkDir / ".project_sha"
+  private def resolveOutputShaPath(nativeWorkDir: os.Path)  = nativeWorkDir / ".output_sha"
+
+  private def fileSha(filePath: os.Path): String = {
+    val md = MessageDigest.getInstance("SHA-1")
+    md.update(os.read.bytes(filePath))
+
+    val digest        = md.digest()
+    val calculatedSum = new BigInteger(1, digest)
+    String.format(s"%040x", calculatedSum)
+  }
+
+  private def projectSha(build: Build.Successful, nativeConfig: sn.NativeConfig) = {
+    val md = MessageDigest.getInstance("SHA-1")
```

Comment:
We may want to hash the Scala CLI version somewhere too. Different Scala CLI versions might have different defaults (for options that are not overridden), so we should probably invalidate the cached binary if it changed.